Updated web_hook_controller.rb to include gitlab@ urls

DavidWylie opened the pull request:
Many Gitlab installations use gitlab@ as a git user.  this was once a default user for gitlab installs.  
Due to this the urls for ssh connections from gitlab can include this user.  
Changing the user is a non trivial event.

This adds gitlab to this list so that it can be used in gocd.

ketan commented:
Since this code is specific to gitlab, it should ideally belong to `git_lab_controller.rb` and not the superclass `web_hook_controller.rb` — as this will affect other webhook implementations. Also, please update the corresponding [spec file](https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/spec/controllers/api/web_hooks/git_lab_controller_spec.rb)

DavidWylie commented:
@ketan Thanks for the feedback.  Ive copied the function to the gitlab webhook controller and added the changes needed and added it to the spec tests.

varshavaradarajan commented:
PR build - https://build.gocd.org/go/pipelines/value_stream_map/trigger/2181

DavidWylie commented:
Thanks.
